crates/bitwarden-crypto/src/keys/encryptable.rs

```diff
@@ -0,0 +1,238 @@
+use super::key_ref::{AsymmetricKeyRef, KeyRef, SymmetricKeyRef};
+use crate::{service::CryptoServiceContext, AsymmetricEncString, CryptoError, EncString};
+
+/// This trait should be implemented by any struct capable of knowing which key it needs
+/// to encrypt or decrypt itself.
+pub trait UsesKey<Key: KeyRef> {
+    fn uses_key(&self) -> Key;
+}
+
+pub trait Encryptable<
+    SymmKeyRef: SymmetricKeyRef,
+    AsymmKeyRef: AsymmetricKeyRef,
+    Key: KeyRef,
+    Output,
+>
+{
+    fn encrypt(
+        &self,
+        ctx: &mut CryptoServiceContext<SymmKeyRef, AsymmKeyRef>,
+        key: Key,
+    ) -> Result<Output, crate::CryptoError>;
+}
+
+pub trait Decryptable<
+    SymmKeyRef: SymmetricKeyRef,
+    AsymmKeyRef: AsymmetricKeyRef,
+    Key: KeyRef,
+    Output,
+>
+{
+    fn decrypt(
+        &self,
+        ctx: &mut CryptoServiceContext<SymmKeyRef, AsymmKeyRef>,
+        key: Key,
+    ) -> Result<Output, crate::CryptoError>;
+}
+
+// Basic Encryptable/Decryptable implementations to and from bytes
```

Comment:
Should these be doc comments?

Reply:
I've mostly added these to split up and organize all the Encryptable/Decryptable implementations.

They all look very similar and when I didn't have these comments is was quite annoying to find one in particular. Maybe what that is telling us is that we should move those impls into separate modules or separate files.